statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,626 @@
+        The variance-covariance of the estimates.
+    pseudo : ndarray
+        The jackknife pseudo-values.
+    """
+    def __init__(self, design, data, cov_method='jack', n_reps=None, center_by='global', bsn=None):
+        super().__init__(design, center_by)
+        self.data = np.asarray(data)
+        if self.data.ndim == 1:
+            self.data = self.data[:, None]
+        if cov_method == "jack":
+            self.est, self.vcov, self.stderr = self._jackknife()
+        else:
+            self.est, self.vcov, self.stderr = self._bootstrap(n_reps, bsn)
+
+    def _stat(self, weights):
+        """
+        Returns calculation of total.
+
+        Parameters
+        ----------
+        weights : np.array
+            The weights used to calculate the total, will either be
+            original design weights or recalculated weights via jk,
+            boot, etc
+
+        Returns
+        -------
+        An array containing the statistic calculated on the columns
+        of the dataset.
+        """
+        return np.dot(weights, self.data)
+
+
+class SurveyQuantile(SurveyStat):
+    """
+    Calculates the quantiles[s] for each column.
+
+    Parameters
+    -------
+    design : SurveyDesign object
+    data : ndarray
+        nxp array of the data to calculate the mean on
+    parameter: array-like
+        array of quantiles to calculate for each column
+
+    Attributes
+    ----------
+    data : ndarray
+        The data which to calculate the quantiles on
+    design :
+        Points to the SurveyDesign object
+    est : ndarray
+        The point estimates of the statistic, calculated on the columns
+        of data.
+    quantile : ndarray
+        The quantile[s] to calculate for each column
+    vcov : ndarray
+        The variance-covariance of the estimates.
+    pseudo : ndarray
+        The jackknife pseudo-values.
+    """
+    def __init__(self, design, data, quantile, cov_method='jack', n_reps=None,
+        center_by='global', bsn=None):
+        self.data = np.asarray(data)
+        if self.data.ndim == 1:
+            self.data = self.data[:, None]
+        super().__init__(design, mse)
+        self.quantile = np.asarray(quantile)
+
+        # give warning if user entered in quantile bigger than one
+        if self.quantile.min() < 0 or self.quantile.max > 1:
+            raise ValueError("quantile[s] should be within [0, 1]")
+        self.n_cw = len(self.design.weights)
+
+        # get quantile[s] for each column
+        self.est = np.empty(self.data.shape[1])
+        # need to call this several times
+        self.stderr = np.empty(self.data.shape[1])
+        if cov_method == "jack":
+            for index in range(self.data.shape[1]):
+                self.est[index], self.stderr[index] = self._jackknife()
+        else:
+            for index in range(self.data.shape[1]):
+                self.est[index], self.stderr[index] = self._bootstrap()
+
+    def _stat(self, weights, col_index):
```

Comment:
`_stat` needs to have a uniform interface (method signature), otherwise there is no way for the parent class to call it properly.  col_index is only present in this subclass.

Also it's confusing that SurveyMean returns an array that contains one mean for every column, whereas SurveyQuantile returns an array that contains many quantiles for one column.